bsc-plugin/src/lib/rooibos/Utils.ts

```diff
@@ -1,147 +1,54 @@
-    let statements = statement.func.body.statements;
-    statements.splice(0, statements.length);
-    let newStatements = (typeof source === 'string') ? getFunctionBody(source) : source;
-    for (let newStatement of newStatements) {
-        statements.push(newStatement);
-    }
-}
-
-export function addOverriddenMethod(file: BrsFile, annotation: AnnotationExpression, target: ClassStatement, name: string, source: string): boolean {
-    let funcSource = `
-  function ${name}()
-    ${source}
-  end function
-  `;
-
-    let tokens = brighterscript.Lexer.scan(funcSource).tokens;
-    let { statements, diagnostics } = brighterscript.Parser.parse(tokens, { mode: brighterscript.ParseMode.BrighterScript });
+import { SourceNode } from 'source-map';
+
+export function overrideAstTranspile(editor: AstEditor, node: Expression | Statement, value: string) {
+    editor.setProperty(node, 'transpile', function transpile(this: Expression | Statement, state) {
+        //indent every line with the current transpile indent level (except the first line, because that's pre-indented by bsc)
+        let source = value.replace(/\r?\n/g, (match, newline) => {
+            return state.newline + state.indent();
+        });
+
+        return [new SourceNode(
+            this.range.start.line + 1,
+            this.range.start.character,
+            state.srcPath,
+            source
+        )];
+    });
+}
+
+export function addOverriddenMethod(file: BrsFile, annotation: AnnotationExpression, target: ClassStatement, name: string, source: string, editor: AstEditor): boolean {
+    let functionSource = `
+        function ${name}()
+            ${source}
+        end function
+    `;
+
+    let { statements, diagnostics } = brighterscript.Parser.parse(functionSource, { mode: brighterscript.ParseMode.BrighterScript });
     let error = '';
     if (statements && statements.length > 0) {
         let statement = statements[0] as FunctionStatement;
         if (statement.func.body.statements.length > 0) {
             let p = brighterscript.createToken(brighterscript.TokenKind.Public, 'public', target.range);
             let o = brighterscript.createToken(brighterscript.TokenKind.Override, 'override', target.range);
             let n = brighterscript.createIdentifier(name, target.range);
-            let cms = new brighterscript.ClassMethodStatement(p, n, statement.func, o);
-            target.body.push(cms);
+            let method = new brighterscript.ClassMethodStatement(p, n, statement.func, o);
+            //bsc has a quirk where it auto-adds a `new` method if missing. That messes with our AST editing, so
+            //trigger that functionality BEFORE performing AstEditor operations. TODO remove this whenever bsc stops doing this.
+            // eslint-disable-next-line @typescript-eslint/dot-notation
+            target['ensureConstructorFunctionExists']?.();
```

Comment:
This can be removed once https://github.com/rokucommunity/brighterscript/pull/586 lands...but it's not hurting anything since we're using the optional chaining for the call...it'll just skip the method alltogether in future bsc versions.